--- flatpak-0.9.4/common/flatpak-oci-registry.c~	2017-05-22 09:02:48.000000000 +0200
+++ flatpak-0.9.4/common/flatpak-oci-registry.c	2017-06-05 10:46:01.316092892 +0200
@@ -264,7 +264,11 @@
   if (fd == -1)
     return NULL;
 
+#ifdef __ILP32__
+  current_etag = g_strdup_printf ("%llu", st_buf.st_mtime);
+#else
   current_etag = g_strdup_printf ("%lu", st_buf.st_mtime);
+#endif
 
   if (etag != NULL && strcmp (current_etag, etag) == 0)
     {
